[nanobind] Nanobind Feature Branch PR [Base] (#1176)
PennyLaneAI/pennylane-lightning: .github/CHANGELOG.md

<h3>Internal changes ⚙️</h3>
  

> - Introduce Nanobind as a new framework for Python bindings, and enhance modularity and performance. 
>   It also include significant improvements to the build system and a first round of codebase simplification. 
>   Legacy PyBind11 code remains accessible for compatibility and will be incrementally phased out.

Reviewer: ```suggestion
- Introduce Nanobind as a new framework for Python bindings, and enhance modularity and performance of binding code. 
  This includes significant improvements to the build system and a first round of codebase simplification. 
  Legacy PyBind11 code remains accessible for compatibility and will be incrementally phased out.
```

Ali Asadi,
Yushao Chen,
Amintor Dusko,
Christina Lee,
Joseph Lee,
Luis Alfredo Nuñez Meneses,
Andrija Paurevic,

> Jake Zaia.

Reviewer 2: To fix the rendering issue:
```suggestion
Jake Zaia.

```
